clash-prelude/src/Clash/Explicit/Prelude.hs

```diff
@@ -242,11 +244,11 @@ window
   -- ^ Window of at least size 1
 window clk rst en x = res
   where
-    res  = x :> prev
-    prev = case natVal (asNatProxy prev) of
+    res  = x :> prev'
+    prev' = case natVal (asNatProxy prev') of
              0 -> repeat def
-             _ -> let next = x +>> prev
-                  in  registerB clk rst en (repeat def) next
+             _ -> let next' = x +>> prev'
+                  in  registerB clk rst en (repeat def) next'
```

Comment:
The fact that you have to do this is a pretty strong hint these names shouldn't be in `Clash.Prelude`. Grepping my projects for `prev` and `next` also reveals a number of uses.

Reply:
That's kind of unavoidable. Every new name we choose might already be picked up in existing user code and I doubt that it's in the interest of the user, if we choose artificially unintuitive names just because of that.

So yes, the user might experience some new warnings about "clashing" names after a Clash update, which are always easy to fix via renaming or hiding the new stuff from `Clash.Prelude`, in case they are not desired.

Please note that finding a good (and intuitive) naming scheme that doesn't clash with basic existing libraries (in particular `base`) is a challenge already, especially if the offered primitives are quire fundamental. I also tried to improve the situation here based on some prior experience I had with the [finite](https://hackage.haskell.org/package/finite-1.4.1.2/docs/Finite.html) library, which comes with a similar interface.

Reply:
I think the point is we write many state machines in Clash and there `prev` and `next` are very intuitive local names. Of course you can't avoid all name clashes, but there are some words that are very common in Clash code and `prev` and `next` are two examples of that specific category. So I agree with Martijn that trying to come up with an alternative is worthwhile in this specific case.

Reply:
before/after?

prior/later?

back/forward?

Did not think them through much, just throwing it out there.

Reply:
`fpred` / `fsucc` for finite versions of Enum functions?